src/components/cards/remote-cards/RemoteCard.tsx

```diff
@@ -0,0 +1,254 @@
+import { CustomColor } from '@/design-system/color/useForegroundColor';
+import { maybeSignUri } from '@/handlers/imgix';
+import { Media } from '@/components/Media';
+import { SheetActionButton } from '@/components/sheet';
+import { colors } from '@/styles';
+
+const GUTTER_SIZE = 40;
+
+const getKeyForLanguage = (key: string, object: any, language: Language) => {
+  if (!object) {
+    return '';
+  }
+
+  const objectOrPrimitive = get(object, key);
+  if (typeof objectOrPrimitive === 'undefined') {
+    return '';
+  }
+
+  if (objectOrPrimitive === null) {
+    return '';
+  }
+
+  if (objectOrPrimitive[language]) {
+    return objectOrPrimitive[language];
+  }
+
+  return objectOrPrimitive[Language.EN_US] ?? '';
+};
+
+type RemoteCardProps = {
+  card: TrimmedCard;
+};
+
+type CardItem = {
+  icon: string;
+  text: string;
+  color?: TextColor | CustomColor;
+};
+
+export const RemoteCard: React.FC<RemoteCardProps> = ({
+  card = {} as TrimmedCard,
+}) => {
+  const { navigate } = useNavigation();
+  const { language } = useAccountSettings();
+  const { width } = useDimensions();
+  const { dismissCard } = useRemoteCardContext();
+
+  const { backgroundColor, primaryButton } = card;
+
+  const onDismissCard = useCallback(() => dismissCard(card.sys.id), [
+    card.sys.id,
+    dismissCard,
+  ]);
+
+  const onPress = useCallback(() => {
+    const internalNavigation = (route: string, options: any) => {
+      navigate(route, options);
+    };
+
+    if (primaryButton && primaryButton.url) {
+      Linking.openURL(primaryButton.url);
+    } else if (primaryButton && primaryButton.route) {
+      internalNavigation(primaryButton.route, primaryButton.props);
+    }
+  }, [navigate, primaryButton]);
+
+  const imageForPlatform = () => {
+    if (card.imageCollection.items.length === 1) {
+      return card.imageCollection.items[0].url;
+    }
+
+    if (IS_IOS) {
+      const image = card.imageCollection.items.find(({ url }) =>
+        /ios/i.test(url)
+      );
+      if (image) {
+        return image.url;
+      }
+
+      return card.imageCollection.items[0].url;
+    } else {
+      const image = card.imageCollection.items.find(({ url }) =>
+        /android/i.test(url)
+      );
+      if (image) {
+        return image.url;
+      }
+
+      return card.imageCollection.items[0].url;
+    }
+  };
+
+  if (!card) {
+    return null;
+  }
+
+  const imageUri = imageForPlatform()
+    ? maybeSignUri(imageForPlatform())
+    : undefined;
+
+  return (
+    <Box
+      padding={card.padding ? { custom: card.padding } : undefined}
+      testID={`remote-card-${card.cardKey}`}
+      overflow="hidden"
+      width={{ custom: width - GUTTER_SIZE }}
+      borderRadius={12}
+      background={(backgroundColor as BackgroundColor) ?? 'sufaceSecondary'}
+    >
+      <Box
+        flexDirection="row"
+        width={{ custom: width - GUTTER_SIZE - Number(card.padding) * 2 }}
```

Comment:
i think you might need to check that `card.padding` is not `undefined` otherwise it could get converted to `NaN`